lib/src/functions_model/search_catalog.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:square_connect/square_connect.dart';
+
+part 'search_catalog.g.dart';
+
+@JsonSerializable()
+class SearchCatalogRequest extends Equatable {
+  const SearchCatalogRequest({
+    this.cursor,
+    this.objectTypes,
+    this.includeDeletedObjects,
+    this.includeRelatedObjects,
+    this.beginTime,
+    this.query,
+    this.limit,
+  });
+
+  /// Converts a [Map] to an [SearchCatalogResponse]
+  factory SearchCatalogRequest.fromJson(Map<String, dynamic> json) =>
+      _$SearchCatalogRequestFromJson(json);
+
+  /// Converts a [SearchCatalogResponse] to a [Map]
+  Map<String, dynamic> toJson() => _$SearchCatalogRequestToJson(this);
+
+  final String? cursor;
+  final List<String>? objectTypes;
+  final bool? includeDeletedObjects;
+  final bool? includeRelatedObjects;
+  final String? beginTime;
+  final CatalogQueryType? query;
```

Comment:
```suggestion
  final CatalogQuery? query;
```

Reply:
Do you want CatalogQueryType renamed to CatalogQuery? 
CatalogQuery is not currently defined.

Reply:
@mtwichel

Reply:
Yes, it is now fixed